contracts/StakingRewardsV2.sol

```diff
@@ -345,61 +338,43 @@ contract StakingRewardsV2 is
         whenNotPaused
         updateReward(_account)
     {
-        uint256 reward = rewards[_account];
-        if (reward > 0) {
-            // update state (first)
-            rewards[_account] = 0;
-
-            // emit reward claimed event and index account
-            emit RewardPaid(_account, reward);
-
-            // transfer token from this contract to the account
-            // as newly issued rewards from inflation are now issued as non-escrowed
-            kwenta.transfer(_to, reward);
-        }
-
-        uint256 rewardUSDC = rewardsUSDC[_account] / PRECISION;
-        if (rewardUSDC > 0) {
-            // update state (first)
-            rewardsUSDC[_account] = 0;
-
-            // emit reward claimed event and index account
-            emit RewardPaidUSDC(_account, rewardUSDC);
-
-            // transfer token from this contract to the account
-            // as newly issued rewards from inflation are now issued as non-escrowed
-            usdc.transfer(_to, rewardUSDC);
-        }
+        _processReward(_account, _to, true);
     }
 
-    /// @notice Get the reward of the given account for compounding.
-    /// @dev Retrieves the Kwenta reward without transferring it, as it will be staked immediately after.
-    function _getRewardCompounding(address _account)
+    /// @notice Process KWENTA and USDC rewards
+    /// @dev transferKwenta is set to false when compounding KWENTA rewards
+    /// @param _account The address of the account to process rewards for
+    /// @param _to The address to transfer rewards to
+    /// @param transferKwenta Boolean flag to determine if Kwenta should be transferred
+    /// @return kwentaReward The amount of Kwenta reward processed
+    function _processReward(address _account, address _to, bool transferKwenta)
         internal
-        whenNotPaused
-        updateReward(_account)
-        returns (uint256 reward)
+        returns (uint256 kwentaReward)
```

Comment:
Doesn't seem you're using kwentaReward outside of _processReward, but for consistency sake you should either return both kwenta and USDC reward as a tuple or return neither.